File: frontend-vue/src/services/admin/AdminPdQnaService.js
import http from "@/utils/http-common";
import LoginHeader from "../login/LoginHeader";

class AdminPdQnaService {
    // 전체조회
    getAll(pdQnaTitle, page, size) {
        return http.get(`/admin/admin-pdqna?pdQnaTitle=${pdQnaTitle}&page=${page}&size=${size}`,{
            headers: LoginHeader()
        });
    }
    // 상세조회
    get(pdQnaId) {
        return http.get(`/admin/admin-pdqna/${pdQnaId}`,{
            headers: LoginHeader()
        });
    }
    // 상품문의등록함수
    create(data) {
        return http.post("/admin/admin-pdqna", data,{
            headers: LoginHeader()
        });
    }
    // 상품문의수정함수
    update(pdQnaId, data) {
        return http.put(`/admin/admin-pdqna/${pdQnaId}`, data,{
            headers: LoginHeader()
        });
    }
    // 상품문의삭제함수
    delete(pdQnaId) {
        return http.delete(`/admin/admin-pdqna/deletion/${pdQnaId}`,{
            headers: LoginHeader()
        });
    }
}

export default new AdminPdQnaService();

File: 01_Backend/backend/src/main/java/org/example/backend/service/admin/AdminPdQnaService.java
package org.example.backend.service.admin;

import org.example.backend.model.entity.admin.AdminPdQna;
import org.example.backend.repository.admin.AdminPdQnaRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class AdminPdQnaService {

    @Autowired
    AdminPdQnaRepository adminPdQnaRepository;

    //    TODO: 전체조회(read)
    public List<AdminPdQna> findAll() {
        List<AdminPdQna> list = adminPdQnaRepository.findAll();
        return list;
    }

    //    페이징 처리
    public Page<AdminPdQna> findAllByAdminPdQnaTitleContaining(String pdQnaTitle,
                                                                  Pageable pageable) {
        Page<AdminPdQna> page
                = adminPdQnaRepository
                .findAllByAdminPdQnaTitleContaining(pdQnaTitle, pageable);
        return page;
    }

    //    상세조회
    public Optional<AdminPdQna> findById(Long pdQnaId) {
        Optional<AdminPdQna> optionalAdminPdQna
                = adminPdQnaRepository.findById(pdQnaId);
        return optionalAdminPdQna;
    }

    //    TODO: 등록(insert),수정(update)
    public AdminPdQna save(AdminPdQna adminPdQna) {
        AdminPdQna adminPdQna2 = adminPdQnaRepository.save(adminPdQna);

        return adminPdQna2;
    }

    //    TODO: 삭제(delete)
    public boolean removeById(Long pdQnaId) {
        if (adminPdQnaRepository.existsById(pdQnaId) == true) {
            adminPdQnaRepository.deleteById(pdQnaId);
            return true;
        } else {
            return false;
        }
    }
}
